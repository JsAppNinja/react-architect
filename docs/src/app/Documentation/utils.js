export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
