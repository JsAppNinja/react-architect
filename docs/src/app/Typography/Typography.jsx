import React, { Component } from 'react';
import PureRenderMixin from 'react-addons-pure-render-mixin';

export default class Typography extends Component {
  constructor(props) {
    super(props);

    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
  }

  static path = 'typography';

  render() {
    return (
      <section className="documentation text-container">
        <h3 className="md-display-1">Material Design Typography</h3>
        <p>
          {'The typography has been built off of the '}
          <a href="https://www.google.com/design/spec/style/typography.html">typography specs</a>.
          There are some helper classes included in the scss.
        </p>
        <p>
          By default, the existing header tags, paragraphs, captions, etc will not be updated with
          these classes.
        </p>
        <ul className="md-list">
          <li className="md-display-4">.md-display-4</li>
          <li className="md-display-3">.md-display-3</li>
          <li className="md-display-2">.md-display-2</li>
          <li className="md-display-1">.md-display-1</li>
          <li className="md-headline">.md-headline</li>
          <li className="md-title">.md-title</li>
          <li className="md-subheading-2">.md-subheading-2</li>
          <li className="md-subheading-1">.md-subheading-1</li>
          <li className="md-body-2">.md-body-2</li>
          <li className="md-body-1">.md-body-1</li>
          <li className="md-caption">.md-caption</li>
        </ul>
      </section>
    );
  }
}
//import React from 'react';
//
//import DisplayReadme from '../DisplayReadme';
//import readme from './README.md';
//
//const Typography = (props) => <DisplayReadme {...props} readme={readme} />;
//Typography.path = 'typography';
//
//export default Typography;
