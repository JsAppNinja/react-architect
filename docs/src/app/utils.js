import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
