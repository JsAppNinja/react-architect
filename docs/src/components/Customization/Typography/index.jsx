import React from 'react';
import Markdown from 'components/Markdown';

import './_styles.scss';

const markdown = `
# Material Design Typography

The typography has been built off of the [typography specs](https://www.google.com/design/spec/style/typography.html).
The default font used in react-md is \`'Roboto'\`. This can be changed by overriding
[$md-font-name](/customization/typography?tab=1#variable-md-font-name) and/or
[$md-font-family](/customization/typography?tab=1#variable-md-font-family) before \`@include react-md-everything\`
or \`@include react-md-typography\`.

The base html tags will be modified by default unless you set the sass variable \`$md-typography-extended\`
to \`false\`. The tags will be implemented as:
- \`h1\` - \`.md-display-1\`
- \`h2\` - \`.md-headline\`
- \`h3\` - \`.md-title\`
- \`h4\` - \`.md-subheading-2\`
- \`h5\` - \`.md-subheading-1\`
- \`h6\` - \`.md-body-2\`
- \`p\` - \`.md-body-1\`
- \`caption\` - \`.md-caption\`


All the class names in order of largest to smallest are:
\`.md-display-4\`, \`.md-display-3\`, \`.md-display-2\`, \`.md-display-1\`, \`.md-headline\`,
\`.md-title\`, \`.md-subheading-2\`, \`.md-subheading-1\`, \`.md-body-2\`, \`.md-body-1\`,
and \`.md-caption\`.

#### Examples
`;

const helperMarkdown = `
If the \`$md-typography-include-utilities\` variable is \`true !default\`, the following utility
class names will be created:

- \`.md-text-left\` - aligns text left
- \`.md-text-center\` - aligns text center
- \`.md-text-right\` - aligns text right
- \`.md-text-justify\`, - aligns text by justify
- \`.md-text-capitalize\` - capitalizes the text
- \`.md-text-uppercase\` - uppercases the text
- \`.md-text-lowercase\` - lowercases the text
- \`.md-text-nowrap\` - does not allow line breaks
- \`.md-text-no-select\` - Does not allow use-select
- \`.md-font-light\` - the material design light font weight (300).
- \`.md-font-regular\` - the material design regular font weight (400).
- \`.md-font-medium\` - the material design medium font weight (500).
- \`.md-font-semibold\` - the semibold font weight used by some Google fonts (600).
- \`.md-font-bold\` - the material design bold font weight (700).

> Font weights can be changed by sass variables

If the \`$md-typography-include-text-container\` variable is \`true !default\`, a \`md-text-container\`
class name will be created to position text in the center of the page and keep the line length within
the \`$md-typography-max-line-length\` value.
`;

const Typography = () => (
  <div className="md-grid">
    <div className="md-cell md-cell--12">
      <section className="md-text-container typography-container">
        <Markdown markdown={markdown} component="article" />
        <h1 className="md-display-4">md-display-4</h1>
        <h2 className="md-display-3">md-display-3</h2>
        <h3 className="md-display-2">md-display-2</h3>
        <h4 className="md-display-1">md-display-1</h4>
        <h5 className="md-headline">md-headline</h5>
        <h6 className="md-title">md-title</h6>
        <h6 className="md-subheading-2">md-subheading-2</h6>
        <h6 className="md-subheading-1">md-subheading-1</h6>
        <p className="md-body-2">
          md-body-2: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis finibus sapien nec
          tellus ornare, quis fringilla nibh sodales. Maecenas blandit leo at sem commodo,
          quis mattis augue bibendum. Orci varius natoque penatibus et magnis dis parturient
          montes, nascetur ridiculus mus. Lorem ipsum dolor sit amet, consectetur adipiscing
          elit. Phasellus convallis efficitur viverra. Quisque nec ipsum ut ex feugiat
          congue. Nam vehicula risus in ex viverra, quis condimentum nulla tincidunt. Fusce
          sed lectus gravida, mattis metus non, dignissim massa. Donec placerat nulla metus.
          Praesent fringilla lectus vel velit congue consequat. Phasellus augue nisi, iaculis
          sit amet neque nec, viverra varius nisi.
        </p>
        <p className="md-body-1">
          md-body-1: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed rhoncus nisl eget nisl
          blandit ornare. Integer id sodales tellus. Phasellus imperdiet mollis laoreet.
          Etiam accumsan lorem ac elit euismod rhoncus quis vel felis. Etiam ex lorem,
          scelerisque at tellus at, rhoncus egestas nulla. Curabitur elementum quis lectus
          vitae porta. Pellentesque dignissim velit ac arcu volutpat euismod. Donec tempus a
          dui in blandit. Curabitur commodo eget risus at fringilla. Aliquam in vestibulum
          mauris. Morbi ultricies purus vitae fringilla iaculis.
        </p>
        <table style={{ width: '100%' }}>
          <caption className="md-caption md-text-left">
            md-caption: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce rutrum tincidunt
            ligula, sed suscipit dolor aliquet at. Nunc quis ex a turpis dapibus aliquam ac
            ut ipsum. Maecenas dictum, felis at sagittis semper, turpis dui lobortis sapien,
            ac gravida turpis turpis vel orci. Fusce mollis tortor et mauris facilisis
            pharetra. Nunc gravida odio non metus pharetra, ac tempus nisl eleifend. Praesent
            odio ex, lobortis sed egestas vel, gravida in justo. Mauris suscipit sodales
            vehicula. Duis non enim scelerisque, venenatis eros eu, scelerisque nulla. Duis
            elementum turpis ipsum, ut imperdiet magna pulvinar sed. Ut luctus lorem non eros
            ullamcorper, ut posuere nulla varius. Nunc lacus leo, euismod at augue aliquet,
            scelerisque facilisis turpis. Nunc diam lacus, accumsan eget blandit id, faucibus
            nec velit. Aliquam sollicitudin hendrerit augue, id facilisis ex dapibus vitae.
            Vestibulum vehicula erat eu eros viverra, eget pretium nisi porttitor. Donec ornare,
            nibh laoreet facilisis viverra, augue elit tincidunt diam, vel aliquam magna quam
            vel nunc. Vivamus mauris justo, elementum non rutrum quis, interdum vitae justo.
          </caption>
        </table>
        <h2>Helper Classes</h2>
        <Markdown markdown={helperMarkdown} component="article" />
      </section>
    </div>
  </div>
);

export default Typography;
