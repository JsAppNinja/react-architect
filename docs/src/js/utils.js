import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
