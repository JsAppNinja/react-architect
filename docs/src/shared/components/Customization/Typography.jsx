import React, { PureComponent } from 'react';
import Markdown from 'components/Markdown';
import loremIpsum from 'lorem-ipsum';

import './_typography.scss';

const markdown = `
# Material Design Typography

The typography has been built off of the [typography specs](https://www.google.com/design/spec/style/typography.html).
The default font used in react-md is \`'Roboto'\`. This can be changed by overriding
[$md-font-name](/customization/typography?tab=1#variable-md-font-name) and/or
[$md-font-family](/customization/typography?tab=1#variable-md-font-family) before \`@include react-md-everything\`
or \`@include react-md-typography\`.

The base html tags will be modified by default unless you set the sass variable \`$md-typography-extended\`
to \`false\`. The tags will be implemented as:
- \`h1\` - \`.md-display-1\`
- \`h2\` - \`.md-headline\`
- \`h3\` - \`.md-title\`
- \`h4\` - \`.md-subheading-2\`
- \`h5\` - \`.md-subheading-1\`
- \`h6\` - \`.md-body-2\`
- \`p\` - \`.md-body-1\`
- \`caption\` - \`.md-caption\`


All the class names in order of largest to smallest are:
\`.md-display-4\`, \`.md-display-3\`, \`.md-display-2\`, \`.md-display-1\`, \`.md-headline\`,
\`.md-title\`, \`.md-subheading-2\`, \`.md-subheading-1\`, \`.md-body-2\`, \`.md-body-1\`,
and \`.md-caption\`.

#### Examples
`;

const helperMarkdown = `
If the \`$md-typography-include-utilities\` variable is \`true !default\`, the following utility
class names will be created:

- \`.md-text-left\` - aligns text left
- \`.md-text-center\` - aligns text center
- \`.md-text-right\` - aligns text right
- \`.md-text-justify\`, - aligns text by justify
- \`.md-text-capitalize\` - capitlizes the text
- \`.md-text-uppercase\` - uppercases the text
- \`.md-text-lowercase\` - lowercases the text
- \`.md-text-nowrap\` - does not allow line breaks
- \`.md-text-no-select\` - Does not allow use-select
- \`.md-font-light\` - the material design light font weight.
- \`.md-font-regular\` - the material design regular font weight.
- \`.md-font-medium\` - the material design medium font weight.
- \`.md-font-bold\` - the material design bold font weight.

If the \`$md-typography-include-text-container\` variable is \`true !default\`, a \`md-text-container\`
class name will be created to position text in the center of the page and keep the line length within
the \`$md-typography-max-line-length\` value.
`;

export default class Typography extends PureComponent {
  render() {
    return (
      <main className="md-grid">
        <div className="md-cell md-cell--12">
          <section className="md-text-container typography-container">
            <Markdown markdown={markdown} component="article" />
            <h1 className="md-display-4">md-display-4</h1>
            <h2 className="md-display-3">md-display-3</h2>
            <h3 className="md-display-2">md-display-2</h3>
            <h4 className="md-display-1">md-display-1</h4>
            <h5 className="md-headline">md-headline</h5>
            <h6 className="md-title">md-title</h6>
            <h6 className="md-subheading-2">md-subheading-2</h6>
            <h6 className="md-subheading-1">md-subheading-1</h6>
            <p className="md-body-2">md-body-2: {loremIpsum({ count: 1, units: 'paragraphs' })}</p>
            <p className="md-body-1">md-body-1: {loremIpsum({ count: 1, units: 'paragraphs' })}</p>
            <table style={{ width: '100%' }}>
              <caption className="md-caption md-text-left">md-caption: {loremIpsum({ count: 1, units: 'paragraphs' })}</caption>
            </table>
            <h2>Helper Classes</h2>
            <Markdown markdown={helperMarkdown} component="article" />
          </section>
        </div>
      </main>
    );
  }
}
