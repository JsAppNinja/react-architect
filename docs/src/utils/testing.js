/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { render, shallow, mount } from 'enzyme';
import rootReducer from 'state';

/**
 * I'm lazy.
 *
 * Could be lazier and do `expectSnapshot` though..
 */
export function createSnapshot(children) {
  return renderer.create(children).toJSON();
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * uses something from react-router.
 */
export function createRouterSnapshot(children, location = '/', context = {}) {
  return createSnapshot(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires the redux provider.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires both the redux Provider and something from react-router.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component relies on react-router. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createRouterSnapshot should
 * be used.
 */
export function renderRouterSnapshot(children, location = '/', context = {}) {
  return render(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires the redux Provider. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createReduxSnapshot should
 * be used.
 */
export function renderReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires both the redux Provider and something from react-router.
 * This should be used when a component * or child component uses findDOMNode or the CSSTransitionGroup,
 * otherwise the createReduxRouterSnapshot should be used.
 */
export function renderReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-router.
 */
export function shallowWithRouter(children, location = '/', context = {}) {
  return shallow(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}


/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-router.
 */
export function mountWithRouter(children, location = '/', context = {}) {
  return mount(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function shallowWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return shallow(<Provider store={store}>{children}</Provider>);
}

/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function mountWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return mount(<Provider store={store}>{children}</Provider>);
}

/**
 * A utility function that will override the console's behavior and throw errors
 * if console.error occurs. This is useful when you want to consider React warnings
 * as failures.
 */
export function captureConsole(match = null) {
  const console = global.console;

  beforeAll(() => {
    global.console = {
      ...console,
      error: (error) => {
        if (match === null || error.match(match)) {
          throw new Error(error);
        }
      },
    };
  });

  afterAll(() => {
    global.console = console;
  });
}
