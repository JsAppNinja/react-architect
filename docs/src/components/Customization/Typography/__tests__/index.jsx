/* eslint-env jest */
import React from 'react';
import { createRouterSnapshot } from 'utils/testing';

import Typography from '../';

describe('Typography', () => {
  it('should render correctly', () => {
    const tree = createRouterSnapshot(<Typography />);
    expect(tree).toMatchSnapshot();
  });
});
