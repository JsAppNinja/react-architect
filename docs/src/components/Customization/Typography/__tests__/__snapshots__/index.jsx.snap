// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly 1`] = `
<div
  className="md-grid"
>
  <div
    className="md-cell md-cell--12"
  >
    <section
      className="md-text-container typography-container"
    >
      <article
        className="markdown-container"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h1 id=\\"material-design-typography\\">Material Design Typography</h1>
          <p>The typography has been built off of the <a href=\\"https://www.google.com/design/spec/style/typography.html\\" class=\\"link\\">typography specs</a>.
          The default font used in react-md is <code>&#39;Roboto&#39;</code>. This can be changed by overriding
          <a href=\\"/customization/typography?tab=1#variable-md-font-name\\" class=\\"link\\">$md-font-name</a> and/or
          <a href=\\"/customization/typography?tab=1#variable-md-font-family\\" class=\\"link\\">$md-font-family</a> before <code>@include react-md-everything</code>
          or <code>@include react-md-typography</code>.</p>
          <p>The base html tags will be modified by default unless you set the sass variable <code>$md-typography-extended</code>
          to <code>false</code>. The tags will be implemented as:</p>
          <ul>
          <li><code>h1</code> - <code>.md-display-1</code></li>
          <li><code>h2</code> - <code>.md-headline</code></li>
          <li><code>h3</code> - <code>.md-title</code></li>
          <li><code>h4</code> - <code>.md-subheading-2</code></li>
          <li><code>h5</code> - <code>.md-subheading-1</code></li>
          <li><code>h6</code> - <code>.md-body-2</code></li>
          <li><code>p</code> - <code>.md-body-1</code></li>
          <li><code>caption</code> - <code>.md-caption</code></li>
          </ul>
          <p>All the class names in order of largest to smallest are:
          <code>.md-display-4</code>, <code>.md-display-3</code>, <code>.md-display-2</code>, <code>.md-display-1</code>, <code>.md-headline</code>,
          <code>.md-title</code>, <code>.md-subheading-2</code>, <code>.md-subheading-1</code>, <code>.md-body-2</code>, <code>.md-body-1</code>,
          and <code>.md-caption</code>.</p>
          <h4 id=\\"examples\\">Examples</h4>
          ",
          }
        }
      />
      <h1
        className="md-display-4"
      >
        md-display-4
      </h1>
      <h2
        className="md-display-3"
      >
        md-display-3
      </h2>
      <h3
        className="md-display-2"
      >
        md-display-2
      </h3>
      <h4
        className="md-display-1"
      >
        md-display-1
      </h4>
      <h5
        className="md-headline"
      >
        md-headline
      </h5>
      <h6
        className="md-title"
      >
        md-title
      </h6>
      <h6
        className="md-subheading-2"
      >
        md-subheading-2
      </h6>
      <h6
        className="md-subheading-1"
      >
        md-subheading-1
      </h6>
      <p
        className="md-body-2"
      >
        md-body-2: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis finibus sapien nec tellus ornare, quis fringilla nibh sodales. Maecenas blandit leo at sem commodo, quis mattis augue bibendum. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus convallis efficitur viverra. Quisque nec ipsum ut ex feugiat congue. Nam vehicula risus in ex viverra, quis condimentum nulla tincidunt. Fusce sed lectus gravida, mattis metus non, dignissim massa. Donec placerat nulla metus. Praesent fringilla lectus vel velit congue consequat. Phasellus augue nisi, iaculis sit amet neque nec, viverra varius nisi.
      </p>
      <p
        className="md-body-1"
      >
        md-body-1: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed rhoncus nisl eget nisl blandit ornare. Integer id sodales tellus. Phasellus imperdiet mollis laoreet. Etiam accumsan lorem ac elit euismod rhoncus quis vel felis. Etiam ex lorem, scelerisque at tellus at, rhoncus egestas nulla. Curabitur elementum quis lectus vitae porta. Pellentesque dignissim velit ac arcu volutpat euismod. Donec tempus a dui in blandit. Curabitur commodo eget risus at fringilla. Aliquam in vestibulum mauris. Morbi ultricies purus vitae fringilla iaculis.
      </p>
      <table
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <caption
          className="md-caption md-text-left"
        >
          md-caption: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce rutrum tincidunt ligula, sed suscipit dolor aliquet at. Nunc quis ex a turpis dapibus aliquam ac ut ipsum. Maecenas dictum, felis at sagittis semper, turpis dui lobortis sapien, ac gravida turpis turpis vel orci. Fusce mollis tortor et mauris facilisis pharetra. Nunc gravida odio non metus pharetra, ac tempus nisl eleifend. Praesent odio ex, lobortis sed egestas vel, gravida in justo. Mauris suscipit sodales vehicula. Duis non enim scelerisque, venenatis eros eu, scelerisque nulla. Duis elementum turpis ipsum, ut imperdiet magna pulvinar sed. Ut luctus lorem non eros ullamcorper, ut posuere nulla varius. Nunc lacus leo, euismod at augue aliquet, scelerisque facilisis turpis. Nunc diam lacus, accumsan eget blandit id, faucibus nec velit. Aliquam sollicitudin hendrerit augue, id facilisis ex dapibus vitae. Vestibulum vehicula erat eu eros viverra, eget pretium nisi porttitor. Donec ornare, nibh laoreet facilisis viverra, augue elit tincidunt diam, vel aliquam magna quam vel nunc. Vivamus mauris justo, elementum non rutrum quis, interdum vitae justo.
        </caption>
      </table>
      <h2>
        Helper Classes
      </h2>
      <article
        className="markdown-container"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>If the <code>$md-typography-include-utilities</code> variable is <code>true !default</code>, the following utility
          class names will be created:</p>
          <ul>
          <li><code>.md-text-left</code> - aligns text left</li>
          <li><code>.md-text-center</code> - aligns text center</li>
          <li><code>.md-text-right</code> - aligns text right</li>
          <li><code>.md-text-justify</code>, - aligns text by justify</li>
          <li><code>.md-text-capitalize</code> - capitalizes the text</li>
          <li><code>.md-text-uppercase</code> - uppercases the text</li>
          <li><code>.md-text-lowercase</code> - lowercases the text</li>
          <li><code>.md-text-nowrap</code> - does not allow line breaks</li>
          <li><code>.md-text-no-select</code> - Does not allow use-select</li>
          <li><code>.md-font-light</code> - the material design light font weight (300).</li>
          <li><code>.md-font-regular</code> - the material design regular font weight (400).</li>
          <li><code>.md-font-medium</code> - the material design medium font weight (500).</li>
          <li><code>.md-font-semibold</code> - the semibold font weight used by some Google fonts (600).</li>
          <li><code>.md-font-bold</code> - the material design bold font weight (700).</li>
          </ul>
          <blockquote>
          <p>Font weights can be changed by sass variables</p>
          </blockquote>
          <p>If the <code>$md-typography-include-text-container</code> variable is <code>true !default</code>, a <code>md-text-container</code>
          class name will be created to position text in the center of the page and keep the line length within
          the <code>$md-typography-max-line-length</code> value.</p>
          ",
          }
        }
      />
    </section>
  </div>
</div>
`;
